lib/streamlit/elements/multiselect.py

```diff
@@ -264,6 +291,16 @@ def _multiselect(
             serializer=serde.serialize,
             ctx=ctx,
         )
+        if max_selections is not None:
+            if get_default_count(widget_state.value) > max_selections:
+                raise StreamlitAPIException(
+                    get_options_message(
+                        get_default_count(widget_state.value), max_selections
+                    )
+                )
+            else:
```

Comment:
We can get rid of this `else` statement since it's effectively dead code at this point: we've either set the value of `max_selections` above or will have errored out

Also, without the `else` statement, the if statements can be flattened into `if max_selections is not None and get_default_count(widget_state.value) > max_selections`.